src/population_loader.rs

```diff
@@ -0,0 +1,140 @@
+use crate::parameters::Parameters;
+use ixa::{
+    context::Context,
+    define_person_property, define_person_property_with_default,
+    error::IxaError,
+    global_properties::ContextGlobalPropertiesExt,
+    people::{ContextPeopleExt, PersonId},
+};
+
+use serde::Deserialize;
+use std::path::PathBuf;
+
+#[derive(Deserialize, Debug)]
+#[allow(non_snake_case)]
+pub struct PeopleRecord<'a> {
+    age: u8,
+    homeId: &'a [u8],
+}
+
+define_person_property!(Age, u8);
+define_person_property!(HomeId, usize);
+define_person_property_with_default!(Alive, bool, true);
+
+define_person_property!(CensusTract, usize);
+
+fn create_person_from_record(
+    context: &mut Context,
+    person_record: &PeopleRecord,
+) -> Result<PersonId, IxaError> {
+    let tract: usize = String::from_utf8(person_record.homeId[..11].to_owned())
+        .expect("Home id should have 11 digits for tract + n digits for home id")
+        .parse()
+        .expect("Could not parse census tract");
+
+    let home_id: usize = String::from_utf8(person_record.homeId.to_owned())
+        .expect("Could not read home id")
+        .parse()
+        .expect("Could not parse home id");
+
+    let person_id = context.add_person((
+        (Age, person_record.age),
+        (HomeId, home_id),
+        (CensusTract, tract),
+    ))?;
+    Ok(person_id)
+}
+
+fn load_synth_population(context: &mut Context, synth_input_file: PathBuf) -> Result<(), IxaError> {
+    let mut reader =
+        csv::Reader::from_path(synth_input_file).expect("Failed to open file. No headers found.");
+    let mut raw_record = csv::ByteRecord::new();
+    let headers = reader.byte_headers().unwrap().clone();
+
+    while reader.read_byte_record(&mut raw_record).unwrap() {
+        let record: PeopleRecord = raw_record
+            .deserialize(Some(&headers))
+            .expect("Failed to parse record.");
+        create_person_from_record(context, &record)?;
+    }
+    Ok(())
+}
+
+pub fn init(context: &mut Context) -> Result<(), IxaError> {
+    let parameters = context
+        .get_global_property_value(Parameters)
+        .unwrap()
+        .clone();
+
+    load_synth_population(context, PathBuf::from(parameters.synth_population_file))?;
+    context.index_property(Age);
+    context.index_property(CensusTract);
```

Comment:
I'm unsure why this index is being created in this fxn. Shouldn't it be in `main`? Which properties we query isn't a property of loading.

Reply:
Whether you want to have an index on a property depends on how you will be querying it, which is a property of the rest of the program, not of the code that loads it, so I think that arguably it should be near the code that is doing the frequent querying.

As an aside, `index_property()` really tells Ixa to index it, but doesn't necessarily cause that index to be computed right away (though I think that's what happens now).